debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -388,6 +376,46 @@ public Lsn getFromLsn() {
                                 columns.get(changeTableObjectId)));
             }
             return changeTables;
+        };
+
+        String query = GET_LIST_OF_CDC_ENABLED_TABLES.replace(DATABASE_NAME_PLACEHOLDER, databaseName);
+        if ((fromLsn.getBinary() == null && toLsn.getBinary() == null)) {
+            return queryAndMap(query, mapper);
+        }
+        else if (fromLsn.getBinary() == null) {
+            query = query + " WHERE ct.start_lsn <= ?";
```

Comment:
These conditions apply to the optimized SQL query implemented in https://github.com/sugarcrm/debezium/pull/61. Later, in [CXP-1156](https://sugarcrm.atlassian.net/browse/CXP-1156) we concluded that the SQL part of the optimization wasn't essential. Fixing the bug where `sys.sp_cdc_help_change_data_capture` would get called on every iteration was sufficient.

Please revert the SQL changes from https://github.com/sugarcrm/debezium/pull/61 and implement the improvement on top of the original implementation. Otherwise, it will have to be done anyways when backporting the fix to the 1:1 branch and contributing it upstream.

Reply:
@morozov As I understand `sys.sp_cdc_help_change_data_capture` returns everything it can without any limits. While the main purpose of my changes is to correctly filter found capture instances according to LSN interval. Do you suggest to do post filtering in the code?

Reply:
No, I thought `sys.sp_cdc_help_change_data_capture` would accept min/max LSN but it won't. This makes it a valid reason to keep our optimization SPROC → SQL in place.